pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +364,355 @@ def Sphere(
+        A monotonically increasing sequence of values specifying radial
+        points. If specified, it overrides any choices for ``outer_radius``,
+        ``inner_radius``, and ``radius_resolution``.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying theta
+        points in degrees. If specified, it overrides any choices for
+        ``start_theta``, ``end_theta``, and ``theta_resolution``.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying phi
+        points in degrees. If specified, it overrides any choices for
+        ``start_phi``, ``end_phi``, and ``phi_resolution``.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees latitude.
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Solid sphere mesh.
+
+    See Also
+    --------
+    pyvista.Sphere
+
+    Examples
+    --------
+    Create a solid sphere.
+
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> solid_sphere = pv.SolidSphere()
+    >>> solid_sphere.plot(show_edges=True)
+
+    A solid sphere is 3D in comparison to the 2d `pyvista.Sphere`.
+    Generate a solid hemi-sphere to see the internal structure.
+
+    >>> isinstance(solid_sphere, pv.UnstructuredGrid)
+    True
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180, end_theta=360
+    ... )
+    >>> partial_solid_sphere.plot(show_edges=True)
+
+    To see the cell structure inside the solid sphere,
+    only 1/4 of the sphere is generated. The cells exploded
+    and colored by radial position.
+
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180,
+    ...     end_theta=360,
+    ...     start_phi=0,
+    ...     end_phi=90,
+    ...     radius_resolution=5,
+    ...     theta_resolution=8,
+    ...     phi_resolution=8,
+    ... )
+    >>> partial_solid_sphere["cell_radial_pos"] = np.linalg.norm(
+    ...     partial_solid_sphere.cell_centers().points, axis=-1
+    ... )
+    >>> partial_solid_sphere.explode(1).plot()
+
+    """
+    if radius is None:
+        radius = np.linspace(inner_radius, outer_radius, radius_resolution)
+    if theta is None:
+        theta = np.linspace(start_theta, end_theta, theta_resolution)
+    if phi is None:
+        phi = np.linspace(start_phi, end_phi, phi_resolution)
+
+    if radius[0] < 0:
+        raise ValueError("minimum radius cannot be negative")
+    if theta[0] < 0:
+        raise ValueError("minimum theta cannot be negative")
+    if phi[0] < 0:
+        raise ValueError("minimum phi cannot be negative")
+    if theta[-1] > 360:
+        raise ValueError("maximum theta cannot be > 360")
+    if phi[-1] > 180:
+        raise ValueError("maximum phi cannot be > 180")
```

Comment:
Let's express that it is float.
```suggestion
    if radius[0] < 0.0:
        raise ValueError("minimum radius cannot be negative")
    if theta[0] < 0.0:
        raise ValueError("minimum theta cannot be negative")
    if phi[0] < 0.0:
        raise ValueError("minimum phi cannot be negative")
    if theta[-1] > 360.0:
        raise ValueError("maximum theta cannot be > 360.0")
    if phi[-1] > 180.0:
        raise ValueError("maximum phi cannot be > 180.0")
```

Reply:
I don't think there is any difference in Python, but it seems fine to me.